test/conversations/Conversation.test.ts

```diff
@@ -660,4 +664,219 @@ describe('conversation', () => {
+        [`${ContentTypeText.authorityId}/${ContentTypeText.typeId}`]: new TextCodec(),
+      }
+
+      const stream = await Stream.create<DecodedMessageExport>(
+        aliceConversation.getClient(),
+        [aliceConversation.topic],
+        async (e) => {
+          const convo = deepcopy(aliceConversation.export())
+          if(convo.version !== 'v1') {
+            throw new Error('unexpected convo version')
+          }
+          const keys = deepcopy(aliceConversation.getClient().legacyKeys)
+          return await decodeMessageV1(
+            e,
+            convo,
+            {codecFor: (c: ContentTypeId) => baseCodecs[`${c.authorityId}/${c.typeId}`]},
+            keys
+          )
+        }
+      )
+
+      await sleep(100)
+      await bobConversation.send('Hi Alice')
+
+      for await (const message of stream) {
+        expect(message.content).toBe('Hi Alice')
+        break
+      }
+
+      let result = await stream.next()
+      expect(result.done).toBeTruthy()
+      await stream.return()
+    })
+
+    it('encodeMessageV1', async () => {
+      const aliceConversation = await alice.conversations.newConversation(
+        bob.address
+      )
+      const bobConversation = await bob.conversations.newConversation(
+        alice.address
+      )
+      const baseCodecs = {
+        [`${ContentTypeText.authorityId}/${ContentTypeText.typeId}`]: new TextCodec(),
+      }
+
+      await sleep(100)
+
+      let topics: string[]
+      let recipient = await bobConversation.getClient().getUserContact(bobConversation.peerAddress)
+      if (!recipient) {
+        throw new Error(`recipient ${bobConversation.peerAddress} is not registered`)
+      }
+      if (!(recipient instanceof PublicKeyBundle)) {
+        recipient = recipient.toLegacyBundle()
+      }
+
+      if (!bobConversation.getClient().contacts.has(bobConversation.peerAddress)) {
+        topics = [
+          buildUserIntroTopic(bobConversation.peerAddress),
+          buildUserIntroTopic(bobConversation.getClient().address),
+          bobConversation.topic,
+        ]
+        bobConversation.getClient().contacts.add(bobConversation.peerAddress)
+      } else {
+        topics = [bobConversation.topic]
+      }
+
+      const keys = deepcopy(bobConversation.getClient().legacyKeys)
+      console.log(bobConversation.getClient().legacyKeys.getPublicKeyBundle().identityKey.getEthereumAddress())
+      console.log(PrivateKeyBundleV1.from(keys).getPublicKeyBundle().identityKey.getEthereumAddress())
+      console.log(bobConversation.getClient().legacyKeys.getPublicKeyBundle().preKey.getEthereumAddress())
+      console.log(PrivateKeyBundleV1.from(keys).getPublicKeyBundle().preKey.getEthereumAddress())
+
+      const msg = await encodeMessageV1(
+        "Hi Alice!",
+        keys,
+        recipient,
+        {codecFor: (c: ContentTypeId) => baseCodecs[`${c.authorityId}/${c.typeId}`]},
+        undefined
+      )
+      console.log(bob.address)
```

Comment:
This test is the broken one, here bob.address and the message sender are somehow not the same...